utilities/xml_parser.py

```diff
@@ -0,0 +1,170 @@
+    xml_files = glob.glob(filepath)
+    passed_tests = []
+    failed_tests = []
+    for xml_file in xml_files:
+        print("-- " + xml_file + " --")
+        doc = xml.dom.minidom.parse(xml_file)
+        testsuitelem = doc.getElementsByTagName("testsuite")
+        for ts in testsuitelem:
+            tsname = ts.getAttribute("name")
+            testcaseelem = ts.getElementsByTagName("testcase")
+            failed = False
+            for tc in testcaseelem:
+                tcname = tc.getAttribute("name")
+                tcerror = tc.getElementsByTagName("error")
+                for tce in tcerror:
+                    failed_tests.append(tcname)
+                    failed = True
+                if not failed:
+                    passed_tests.append(tcname)
+
+    if failed_tests:
+        failed_tests = transform_and_write_to_file(failed_tests, "failed_tests.conf")
+        print(" or ".join(failed_tests))
+
+    if passed_tests:
+        passed_tests = transform_and_write_to_file(passed_tests, "passed_tests.conf")
+
+
+def parse_testreport_result_xml(filepath=""):
+    if filepath.startswith("http://") or filepath.startswith("https://"):
+        if filepath.endswith(".xml"):
+            url_path = filepath
+        else:
+            url_path = filepath + "/testReport/api/xml?pretty=true"
+        jobnamebuild = filepath.split('/')
+        if not os.path.exists('logs'):
+            os.mkdir('logs')
+        newfilepath = 'logs' + ''.join(os.sep) + '_'.join(jobnamebuild[-3:]) + "_testresult.xml"
+        # print("Downloading " + url_path + " to " + newfilepath)
+        try:
+            filedata = urllib.request.urlopen(url_path)
+            datatowrite = filedata.read()
+            filepath = newfilepath
+            with open(filepath, 'wb') as f:
+                f.write(datatowrite)
+        except Exception as ex:
+            print("Error:: " + str(ex) + "! Please check if " +
+                  url_path + " URL is accessible!!")
+            print("Running all the tests instead for now.")
+            return None, None
+    if filepath == "":
+        filepath = "logs/**/*.xml"
+    xml_files = glob.glob(filepath)
+    passed_tests = []
+    failed_tests = []
+    for xml_file in xml_files:
+        doc = xml.dom.minidom.parse(xml_file)
+        testresultelem = doc.getElementsByTagName("testsuites")
+        testsuitelem = testresultelem[0].getElementsByTagName("testsuite")
+        for ts in testsuitelem:
+            testcaseelem = ts.getElementsByTagName("testcase")
+            for tc in testcaseelem:
+                tcname = tc.getAttribute("name")
+                if tc.getElementsByTagName("failure") or tc.getElementsByTagName("error"):
+                    failed = True
+                    failed_tests.append(tcname.split('[')[0])
+                else:
+                    failed = False
+                    passed_tests.append(tcname)
+
+    if failed_tests:
```

Comment:
is this condition required as this is always be true. Though it is empty, it is still treated as true. May be you want to add the condtion that length of failed tests is more than 0